### PR TITLE
K8s: handle multiple versions of the same group in standalone mode

### PR DESCRIPTION
**Pull request by charandas**
**What is this feature?**

Moves up the place where apiGroupInfo is instantiated so that the caller can manage concerns related to registering api groups of the same group name but different versions.

**Why do we need this feature?**

We are getting to the point where we need to deploy at least one api group with a new version while maintaining backwards compat in order to not disturb existing deployments.

**Who is this feature for?**

Grafana App Platform

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/apiserver/builder/common.go [hunk 1]
```diff
@@ -24,13 +23,19 @@ type APIGroupBuilder interface {
...
+	// where separate API Group Info for the same group (different versions) aren't handled well by
+	// the InstallAPIGroup facility of genericapiserver. Also, we can only ever call InstallAPIGroup
+	// once on the genericapiserver per group, or we run into double registration startup errors.
+	//
+	// The caller should share the apiGroupInfo passed into this function across builder versions of the same group.
+	// UpdateAPIGroupInfo builds the group+version behavior updating the passed in apiGroupInfo in place
+	UpdateAPIGroupInfo(
+		apiGroupInfo *genericapiserver.APIGroupInfo,
 		scheme *runtime.Scheme,
-		codecs serializer.CodecFactory,
```
Comment (charandas) [inline pkg/services/apiserver/builder/common.go:30]: `codecs` is not necessary in any iteration of the new usage once apiGroupInfo itself is moved to being created outside the Update func here.

**Review by toddtreece — APPROVED** [review]

**Comment by charandas** [comment]
/deploy-to-hg --enterprise-ref "charandas/cloud-config-namespaced+legacy"

**Comment by charandas** [comment]
/deploy-to-hg --enterprise-ref 1fde757e57656003d730e89d685e4b4f88f94d4e

**Pull request merged**